tests/integration/cartography/intel/aws/test_lambda.py

```diff
@@ -179,131 +239,194 @@ def test_load_lambda_function_aliases_relationships(neo4j_session):
+
+def test_load_lambda_event_source_mappings(neo4j_session):
+    # Create Test AWSAccount first for sub-resource relationships
+    neo4j_session.run(
         """
-        MATCH (n1:AWSLambda)-[:KNOWN_AS]->(n2:AWSLambdaFunctionAlias) RETURN n1.id, n2.id;
+        MERGE (aws:AWSAccount{id: $aws_account_id})
+        ON CREATE SET aws.firstseen = timestamp()
+        SET aws.lastupdated = $aws_update_tag
         """,
+        aws_account_id=TEST_ACCOUNT_ID,
+        aws_update_tag=TEST_UPDATE_TAG,
     )
-    actual = {(r["n1.id"], r["n2.id"]) for r in result}
-
-    assert actual == expected_nodes
-
 
-def test_load_lambda_event_source_mappings(neo4j_session):
     data = tests.data.aws.lambda_function.LIST_EVENT_SOURCE_MAPPINGS
 
-    cartography.intel.aws.lambda_function._load_lambda_event_source_mappings(
+    cartography.intel.aws.lambda_function.load_lambda_event_source_mappings(
         neo4j_session,
         data,
+        TEST_ACCOUNT_ID,
         TEST_UPDATE_TAG,
     )
 
-    expected_nodes = {
-        "i01",
-        "i02",
+    # Test Lambda event source mapping nodes were created correctly
+    assert check_nodes(neo4j_session, "AWSLambdaEventSourceMapping", ["id"]) == {
+        ("i01",),
+        ("i02",),
     }
 
-    nodes = neo4j_session.run(
+    # Test AWSAccount -> AWSLambdaEventSourceMapping RESOURCE relationships
+    assert check_rels(
+        neo4j_session,
+        "AWSAccount",
+        "id",
+        "AWSLambdaEventSourceMapping",
+        "id",
+        "RESOURCE",
+        rel_direction_right=True,
+    ) == {
+        (TEST_ACCOUNT_ID, "i01"),
+        (TEST_ACCOUNT_ID, "i02"),
+    }
+
+
+def test_load_lambda_event_source_mappings_relationships(neo4j_session):
+    # Create Test AWSAccount first
+    neo4j_session.run(
         """
-        MATCH (r:AWSLambdaEventSourceMapping) RETURN r.id;
+        MERGE (aws:AWSAccount{id: $aws_account_id})
+        ON CREATE SET aws.firstseen = timestamp()
+        SET aws.lastupdated = $aws_update_tag
         """,
+        aws_account_id=TEST_ACCOUNT_ID,
+        aws_update_tag=TEST_UPDATE_TAG,
     )
 
-    actual_nodes = {n["r.id"] for n in nodes}
-
-    assert actual_nodes == expected_nodes
-
-
-def test_load_lambda_event_source_mappings_relationships(neo4j_session):
     # Create Test Lambda Function
     data = tests.data.aws.lambda_function.LIST_LAMBDA_FUNCTIONS
 
+    # Transform the data first
+    transformed_data = cartography.intel.aws.lambda_function.transform_lambda_functions(
+        data,
+        TEST_REGION,
+    )
+
     cartography.intel.aws.lambda_function.load_lambda_functions(
         neo4j_session,
-        data,
+        transformed_data,
         TEST_REGION,
         TEST_ACCOUNT_ID,
         TEST_UPDATE_TAG,
     )
 
     esm = tests.data.aws.lambda_function.LIST_EVENT_SOURCE_MAPPINGS
 
-    cartography.intel.aws.lambda_function._load_lambda_event_source_mappings(
+    cartography.intel.aws.lambda_function.load_lambda_event_source_mappings(
```

Comment:
Nit: avoid renaming the existing function so that it's clearer to show that your refactor PR doesn't break existing tests